tfs/src/main/java/hudson/plugins/tfs/model/AbstractHookEvent.java

```diff
@@ -115,52 +118,78 @@ static JSONObject fromResponseContributors(final List<GitStatus.ResponseContribu
-                    {
+
+                if (job instanceof WorkflowJob) {
+                    final FlowDefinition jobDef = ((WorkflowJob) job).getDefinition();
+                    if (jobDef instanceof CpsScmFlowDefinition) {
                         final SCM jobSCM = ((CpsScmFlowDefinition) jobDef).getScm();
-                        if (jobSCM instanceof GitSCM)
-                        {
-                            final GitSCM gitJobSCM = (GitSCM)jobSCM;
+                        if (jobSCM instanceof GitSCM) {
+                            final GitSCM gitJobSCM = (GitSCM) jobSCM;
                             final URIish uri = gitCodePushedEventArgs.getRepoURIish();
 
                             for (final UserRemoteConfig remoteConfig : gitJobSCM.getUserRemoteConfigs()) {
                                 final String jobRepoUrl = remoteConfig.getUrl();
-                                
-                                if (StringUtils.equals(jobRepoUrl, uri.toString())) {
+
+                                if (StringUtils.equalsIgnoreCase(jobRepoUrl, uri.toString())) {
                                     shouldRun = true;
                                     break;
                                 }
-                            }                            
+                            }
                         }
                     }
                 }
-                
-                TeamPushTrigger pushTrigger = TeamEventsEndpoint.findTrigger(job, TeamPushTrigger.class);               
-                if (shouldRun && pushTrigger == null)
-                    pushTrigger = new TeamPushTrigger(job);                            
+
+                TeamPushTrigger pushTrigger = TeamEventsEndpoint.findTrigger(job, TeamPushTrigger.class);
+                if (shouldRun && pushTrigger == null) {
+                    pushTrigger = new TeamPushTrigger(job);
+                }
 
                 if (pushTrigger != null) {
                     pushTrigger.execute(gitCodePushedEventArgs, actions, bypassPolling);
                     if (bypassPolling) {
                         return new TeamEventsEndpoint.ScheduledResponseContributor(project);
-                    }
-                    else {
+                    } else {
                         return  new TeamEventsEndpoint.PollingScheduledResponseContributor(project);
                     }
                 }
+
+                if (gitCodePushedEventArgs instanceof PullRequestMergeCommitCreatedEventArgs) {
+                    setPullRequestStatus((PullRequestMergeCommitCreatedEventArgs) gitCodePushedEventArgs, GitStatusState.Failed, "Fail to launch Jenkins CI build", targetUrl);
+                }
             }
         }
+
+        return null;
+    }
+
+    public TeamGitStatus setPullRequestStatus(final PullRequestMergeCommitCreatedEventArgs gitCodePushedEventArgs, final GitStatusState buildState, final String buildDescription, final String targetUrl) {

```

Comment:
Again, it would be nice for this to live on the JenkinsEventNotifier class.

Reply:
Good suggestion. Refactored this API into JenkinsEventNotifier class, and made it shared by both callers.